main/src/cgeo/geocaching/connector/gc/GCParser.java

```diff
@@ -1328,10 +1332,33 @@ static boolean addToWatchlist(final Geocache cache) {
         } else {
             Log.e("GCParser.addToWatchlist: cache is not on watchlist");
         }
+        // WatchListCount
+        final String watchListPage = GCLogin.getInstance().postRequestLogged(cache.getLongUrl(), null);
+        cache.setWatchlistCount(getWatchListCount(watchListPage));
         return guidOnPage; // on watchlist (=added) / else: error
     }
 
     /**
+     * This method extracts the amount of people watching on a geocache out of the HTMl website passed to it
+     * @param page Page containing the information about howm many people watching on geocache
+     * @return Number of people watching geocache, -1 when error
+     */
+    static int getWatchListCount(final String page)
+    {
+        int count;
+        String sCount = TextUtils.getMatch(page, GCConstants.PATTERN_WATCHLIST_COUNT, true, 1, "notFound", false);
+        if (sCount.equals("notFound")) return -1;
+        try {
+            count = Integer.parseInt(sCount);
```

Comment:
Just return the result from `parseInt` from here, no need for the `count` variable.